### PR TITLE
Fix `cmd/root_test.go` test runs from linked Git worktrees.

### PR DESCRIPTION
**Pull request by mr0re1**
Fix `cmd/root_test.go` test runs from linked Git worktrees.
```
$git worktree add wt && cd wt
$make tests
...
PANIC: root_test.go:112: MySuite.TestCheckGitHashMismatch
```

- Don't use real repo as a subject for inspection, use test ones;
- Use `Check` instead of `Assert`.

**Testing**:
```
$ cd ~/wp/hpc-toolkit && git checkout wt
$ make tests
...
$ git checkout develop && git worktree add wt && cd wt
$ make tests
...
```

**NOTE**:
While running `make tests` from linked worktree, incorrect warning is shown:
```
WARNING: ghpc binary was built from a different commit (wt/7812ced) than the current git branch in ~/wp/hpc-toolkit (develop/edb0ce3). You can rebuild the binary by running 'make'
```
This issue should be addressed in separate PR.

### Submission Checklist

* [x] Have you installed and run this change against pre-commit? (`pre-commit
  install`)
* [x] Are all tests passing? (`make tests`)
* [x] Have you written unit tests to cover this change?
* [x] Is unit test coverage still above 80%?
* [ ] Have you updated all applicable documentation?
* [x] Have you followed the guidelines in our Contributing document?

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -157,10 +158,13 @@ func hpcToolkitRepo() (repo *git.Repository, dir string, err error) {
...
-	if err == nil && isHpcToolkitRepo(*repo) {
-		return repo, dir, err
+	if err != nil {
+		return
+	}
+	if isHpcToolkitRepo(*repo) {
+		return
 	}
-	return nil, "", err
+	return nil, "", errors.New("Not the hpc-toolkit repo")
```
Comment (cboneti) [inline cmd/root.go:167]: This message may not be very clear to an user. Do we not expect it to be ever reached? otherwise, could we try to have something more clear, like, `Unable to find the hpc-toolkit git repo in the file system.` or something of the sort.

Comment (mr0re1) [reply cmd/root.go]: Addressed

File: cmd/root_test.go [hunk 2]
```diff
@@ -19,157 +19,196 @@ package cmd
...
+	if err = os.Chdir(path); err != nil {
+		c.Fatal(err)
+	}
+
 	head, _ := repo.Head()
 	hash := head.Hash().String()
 	branch := head.Name().Short()
 
-	// verify current working directory git hash against hash of v1.0.0
-	mockInjectedGitVariables()
```
Comment (nick-stroud) [inline cmd/root_test.go:122]: nit: Technically we are now missing this case that was previously covered. 

The case being: 
State: The binary was built from a hash that is actually in the git history
Result: Identify mismatch

This is partially covered by the "Baked commit hash doesn't match" case which is testing
State: The binary is built from a hash not in the git repo (random)
Result: Identify mismatch

I don't know if I really care too much about this case as we know we don't search through the git history, maybe consider how hard it would be to add it. You would have to know a valid commit in the git history that is not the initial commit.

File: cmd/root_test.go [hunk 3]
```diff
@@ -19,157 +19,196 @@ package cmd
...
+		c.Check(mismatch, Equals, false)
+	}
 
-	// verify current working directory git hash against random initial hash
-	mockInjectedGitVariables()
-	GitInitialHash = randomGitHash
-	mismatch, b, h, dir = checkGitHashMismatch()
-	c.Assert(mismatch, Equals, false)
-	c.Assert(b, Equals, "")
-	c.Assert(h, Equals, "")
```
Comment (nick-stroud) [inline cmd/root_test.go:135]: nit: Before we verified that in the non-mismatch case we don't emit any spurious data. There is nothing binding us to that contract but I kind of like it in these named return functions which make it easy for random data to slip out.

File: cmd/root_test.go [hunk 4]
```diff
@@ -19,157 +19,196 @@ package cmd
...
 
-	// verify current working directory git hash against hash of v1.0.0
-	mockInjectedGitVariables()
-	mismatch, b, h, dir := checkGitHashMismatch()
-	c.Assert(mismatch, Equals, true)
-	c.Assert(dir, Equals, workDir)
-	c.Assert(b, Equals, branch)
-	c.Assert(h, Equals, hash)
+	{ // Matches
+		GitCommitHash = hash
```
Comment (nick-stroud) [inline cmd/root_test.go:145]: For the first 3 of the 4 cases outlined here logic is depending on `GitCommitHash` and `GitInitialHash` as opposed to just setting the one needed for the case I think the tests would be more readable if you set both at the top of each case. So for example on the "Matches" case you would start with:
```
GitCommitHash = hash
GitInitialHash = init.String()
```

File: cmd/root_test.go (hunk 3 above)
Comment (nick-stroud) [reply cmd/root_test.go:135]: I see this modality several places, I could be convinced otherwise. Thoughts?

File: cmd/root_test.go (hunk 2 above)
Comment (mr0re1) [reply cmd/root_test.go:122]: Added mismatch-case with
```
GitInitialHash = init.String()
GitCommitHash = init.String()
```

File: cmd/root.go [hunk 5]
```diff
@@ -157,10 +158,13 @@ func hpcToolkitRepo() (repo *git.Repository, dir string, err error) {
 	dir = filepath.Dir(e)
 
 	repo, err = git.PlainOpen(dir)
-	if err == nil && isHpcToolkitRepo(*repo) {
-		return repo, dir, err
+	if err != nil {
+		return
```
[the remark this answers is not in the document]
Comment (nick-stroud) [reply cmd/root.go]: I see, I missed that.
In that case I would suggest a different error message that is specific to this case. Maybe:

 >_"ghpc executable found in a git repo other than the hpc-toolkit git repo"_

I would guess @cboneti's suggestion was with the assumption that this was a summarizing error message that was meant to be seen by the user., not a specific case that had no system error.

**Review by nick-stroud — APPROVED** [review]

File: cmd/root.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (mr0re1) [reply cmd/root.go]: Thank you, I was concerned about message being too wordy/specific, but you have found the right words )
